presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorIntegrationSmokeTest.java

```diff
@@ -329,6 +329,80 @@ public void testShardsSystemTable()
     }
 
     @Test
+    public void testShardsSystemTableWithTemporalColumn()
+            throws Exception
+    {
+        // Make sure we have rows in the selected range
+        assertEquals(computeActual("SELECT count(*) >= 1 FROM orders WHERE orderdate BETWEEN date '1992-01-01' AND date '1992-02-08'").getOnlyValue(), true);
+
+        // Create a table that has DATE type temporal column
+        assertUpdate("CREATE TABLE test_shards_system_table_date_temporal\n" +
+                        "WITH (temporal_column = 'orderdate') AS\n" +
+                        "SELECT orderdate, orderkey\n" +
+                        "FROM orders\n" +
+                        "WHERE orderdate BETWEEN date '1992-01-01' AND date '1992-02-08'",
+                "SELECT count(*)\n" +
+                        "FROM orders\n" +
+                        "WHERE orderdate BETWEEN date '1992-01-01' AND date '1992-02-08'"
+        );
+
+        // Create a table that has TIMESTAMP type temporal column
+        assertUpdate("CREATE TABLE test_shards_system_table_timestamp_temporal\n" +
+                        "WITH (temporal_column = 'ordertimestamp') AS\n" +
+                        "SELECT CAST (orderdate AS TIMESTAMP) AS ordertimestamp, orderkey\n" +
+                        "FROM test_shards_system_table_date_temporal",
+                "SELECT count(*)\n" +
+                        "FROM orders\n" +
+                        "WHERE orderdate BETWEEN date '1992-01-01' AND date '1992-02-08'");
+
+        // For table with DATE type temporal column, min/max_timestamp columns must be null while min/max_date columns must not be null
+        assertEquals(computeActual("" +
+                "SELECT count(*)\n" +
+                "FROM system.shards\n" +
+                "WHERE table_schema = 'tpch'\n" +
+                "AND table_name = 'test_shards_system_table_date_temporal'\n" +
+                "AND NOT \n" +
+                "(min_timestamp IS NULL AND max_timestamp IS NULL\n" +
+                "AND NOT min_date IS NULL AND NOT max_date IS NULL)").getOnlyValue(), 0L);
```

Comment:
Nit: write `x IS NOT NULL` rather than `NOT x IS NULL` as that's the canonical way to do it, so it's easier to read

Reply:
Done.